### PR TITLE
Added Haskell num digits

### PR DESCRIPTION
**Pull request by kefirski**
Fixes issue #1815

Changes: Added haskell implementation of num digits

### CODE REVIEW

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

File: code/mathematical-algorithms/basic/Greatest_digit_in_number/Greatest_digit_in_number.hs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+greatest :: Integer -> Integer
+greatest n
+	| modOfTen == n = n
+	| otherwise     = modOfTen `max` greatest quotOfTen
+	where modOfTen  = mod n 10
+	      quotOfTen = quot n 10
```
Comment (Croolis) [inline code/mathematical-algorithms/basic/Greatest_digit_in_number/Greatest_digit_in_number.hs]: >No newline at end of file
Please fix

File: code/mathematical-algorithms/factorial/factorial.scala [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+object Factorial {
+    def apply(n: Int): Int = (1 to n).product
+}
```
Comment (Croolis) [inline code/mathematical-algorithms/factorial/factorial.scala]: >No newline at end of file
Please fix

File: code/mathematical-algorithms/count_digits/count_digits.hs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+countDigits :: Integer -> Integer
+countDigits 0 = 0
+countDigits n = succ $ countDigits $ quot n 10
+
```
Comment (Croolis) [inline code/mathematical-algorithms/count_digits/count_digits.hs]: Please remove

**Pull request merged**